app/controllers/tests_controller.rb

```diff
@@ -2,30 +2,39 @@ class TestsController < ApplicationController
   skip_before_action :check_admin, except: [:index]
 
   def create
-    course = Problem.find(params[:problem_id]).course
+    problem = Problem.includes(assignment: :course).find(params[:problem_id])
+    course = problem.course
     if !current_user_in_course?(course)
       head :forbidden
       return
     end
 
     uploaded_file = params[:file]
 
+    test = Test.new(
```

Comment:
I feel like I've made a similar comment before, but can't find where I've done it; why do we instantiate the Test outside of the transaction with `new` instead of `create` inside the transaction?